spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAddFilesProcedure.java

```diff
@@ -819,6 +842,27 @@ public void testPartitionedImportFromEmptyPartitionDoesNotThrow() {
               RowFactory.create(4, "Will Doe", "facilities", toDate("2021-01-02"))),
           new StructType(dateStruct)).repartition(2);
 
+  private static final StructField[] timeStampstruct = {
+      new StructField("id", DataTypes.IntegerType, true, Metadata.empty()),
+      new StructField("name", DataTypes.StringType, true, Metadata.empty()),
+      new StructField("dept", DataTypes.StringType, true, Metadata.empty()),
+      new StructField("ts", DataTypes.TimestampType, true, Metadata.empty())
+  };
+
+  private static Timestamp toTimestamp(String value) {
```

Comment:
Okay, I see. I was wrong before. This is actually producing a Timestamp because that's what Spark's public row API uses. Instead, could you use SparkSQL and literal values to avoid doing this conversion manually? I think that will make for a much more reliable test.